ec/ecresource/elasticsearchkeystoreresource/expanders.go

```diff
@@ -40,7 +39,7 @@ func expandModel(d *schema.ResourceData) *models.KeystoreContents {
 	return &models.KeystoreContents{
 		Secrets: map[string]models.KeystoreSecret{
 			secretName: {
-				AsFile: ec.Bool(d.Get("as_file").(bool)),
+				AsFile: ec.Bool(state.AsFile.Value),
```

Comment:
I think we should either keep this as is as this is an implementation detail. Or remove all usages of `ec.Bool()`, `ec.String()`, ….

Reply:
IMO `ec.Bool` is a helper method that facilitates passing literals when the receiver is a pointer. There is no sense to use it outside of this case. Also currently these methods are not used everywhere (e.g. https://github.com/elastic/terraform-provider-ec/blob/master/ec/ecresource/deploymentresource/elasticsearch_expanders.go#L299, https://github.com/elastic/terraform-provider-ec/blob/master/ec/ecresource/deploymentresource/elasticsearch_expanders.go#L211 - just a brief search).
IMO it makes sense to use them when there is a reason for that, not just because it was used in some other places (that also could happen by mistake).

Reply:
The receiver is a pointer:
```
	AsFile *bool `json:"as_file,omitempty"`
```

Reply:
Ah. I over-read the "literals" part.

Reply:
Will merge as is and change in a follow-up PR because I'm too lazy to rebase all my other PRs.